packages/helpers/src/participant-states.ts

```diff
@@ -52,12 +53,19 @@ export const getParticipantStates = (channel: ChannelData, forceId: string | und
 
   const {
     isParticipant,
-    participatingRole,
+    participatingRoles,
     allRolesIncluded
   }: CheckParticipantStates = checkParticipantStates(channel, forceId, role, isObserver)
 
-  if (participatingRole) {
-    chosenTemplates = participatingRole.templates
+  if (isParticipant) {
```

Comment:
You've done really well with this logic @lilitkarapetyan - there's some complex stuff going on.

In the next block, it "looks like" `chosenTemplates` will contain the templates for all of the participating roles in the channel.  I _think_ it should only contain the templates for _my_ role.  So, `Blue` could have some templates and `White` could have others.

Reply:
No, I was mistaken.  The chosenTemplates is the union of the ways this role can participate in the channel, which is correct :-)